solr/core/src/java/org/apache/solr/search/VectorSimilaritySourceParser.java

```diff
@@ -110,14 +146,38 @@ private ValueSource createSimilarityFunction(
     }
   }
 
-  private void requireVectorType(FieldType fieldType) throws SyntaxError {
-    if (!(fieldType instanceof DenseVectorField)) {
+  private DenseVectorField requireVectorType(final SchemaField field) throws SyntaxError {
+    final FieldType fieldType = field.getType();
+    if (fieldType instanceof DenseVectorField) {
+      return (DenseVectorField) field.getType();
+    }
+    throw new SolrException(
+        BAD_REQUEST,
+        String.format(
+            Locale.ROOT,
+            "Type mismatch: Expected [%s], but found a different field type for field: [%s]",
+            DenseVectorField.class.getSimpleName(),
+            field.getName()));
+  }
+
+  /**
+   * Helper method that returns the correct Enum instance for the <code>arg</code> String, or throws
+   * a {@link ErrorCode#BAD_REQUEST} with specifics on the "Invalid argument"
+   */
+  private static <T extends Enum<T>> T enumValueOrBadRequest(
```

Comment:
I'm considering whether it's necessary for this function to be generic, especially since it's currently only being used for a single enum class and is marked as private.